### ISSUE TITLE
finish the gazer boss's fight mechanics

### ISSUE THREAD

**Issue by rkettering** [frogatto/frogatto#292]
Adding a checklist of core items to finish off here:
- [x] add a real set of hitpoints and damage resistances
- [x] make the gazerlet formation always be outside the ring of platforms, not inside, except during the spiral
- [x] make an explosion animation for the gazerlets
- [ ] make the gazerlets explode after the laser 'charges' them up - but only once the whole laser mode finishes
- [x] make the gazerlets explode once they (during their aggro state) feel like they're close enough to the player.
- [ ] Add an anura engine feature to allow rotated hitboxes for object collisions
- [x] Create some slight delay before the gazerlets actually give chase once hostile to the player. 

Nice to haves:
- [x] figure out a better way, graphically, to do the mook spawning
- [ ] rather than randomly picking a gazerlet for aggro, instead make the boss actually turn to one of them, and focus a "empowering beam" on it to give the player a cue that it's the one that's going to go.
- [x] make the gazer boss speed up his rate of fire for the wobbly shots as he goes into critical health.
- [x] should the laser attack occur at lower health?
- [x] a mode 7-like background during the boss fight

**Comment by Phanterm** [comment]
Suggestions:

-Gazer boss hitbox perhaps limited to the eyeball only?
-I will try and mess with the scatter shot code to explode outward into smaller shots 
-Giant laser attack occurs ~35-50% health? Seems like a desperation attack.
-perhaps the 'empowerment beam' attack could be a transparent cone that fans out instead of a straight beam, to distinguish itself from the laser.

**Comment by rkettering** [comment]
We've added a few of these things, with twists on some of the ideas.  The boss feels complete now.
